lib/iris/plot.py

```diff
@@ -256,17 +274,87 @@ def _invert_yaxis(v_coord, axes=None):
             axes.invert_yaxis()
 
 
+def _check_contiguity_and_bounds(coord, data, abs_tol=1e-4, transpose=False):
+    """
+    Check that the discontinuous bounds occur where the data is masked.
+
+    If discontinuity occurs but data is masked, raise warning
+    If discontinuity occurs and data is NOT masked, raise error
+
+    Args:
+        coords:
+            Array of the bounds of a 2D coord, of shape (X,Y,4)
+        data:
+            data of the the cube we are plotting
+        abs_tol:
+            tolerance when checking the contiguity
+
+    """
+    if transpose:
+        bounds = coord.bounds.T
+    else:
+        bounds = coord.bounds
```

Comment:
I do find this really weird. I would have expected it to be the other way around!